File: Immigration.py
ㅔㄱ# n: 사람수, times[] : 심사관별 검사 시간
def solution(n, times):
    start = 1
    end = n*max(times)
    answer = end

    while (start <= end):
        mid = (start + end) // 2

        people = 0
        for time in times:
            people += (mid // time)

        if people < n:
            start = mid + 1
        else:
            end = mid - 1
            answer = min(answer, mid)

    return answer


solution(6, {7, 10})

File: src/main/java/cote/programmers/Immigration.java
package cote.programmers;

public class Immigration {
    public long solution(int n, int[] times) {
        int start = 1;      // 최소 시간
        int end = 0;        // 최대 시간
        for (int time : times) {
            end = Math.max(end, time * n);
        }
        int answer = end;     // 최소 시간 업데이트용 변수
        int mid = 0;        // 탐색용 중간값

        while (start <= end) {
            mid = (start + end) / 2;

            int people = 0;     // 각 심사관이 심사할 수 있는 사람의 수
            for (int time : times) {
                people += mid / time;
            }

            if (people < n) {
                start = mid + 1;
            } else {
                end = mid - 1;
                answer = Math.min(answer, mid);
            }
        }

        return answer;
    }
}
